### DISCUSSION TITLE
JavaScript API to pause/play the animation

### DISCUSSION THREAD

**Discussion by angel-herraez** [ruffle-rs/ruffle#3198] [Q&A] [answered]
Hi, I just discovered Ruffle and I am enthusiastic with the chance to reliven some old animations. Thank you for making it!
I have an swf animation that runs in steps. When it stops (pauses) after a step, I was using a button in the webpage to send Flash the Play() method.
I wonder if this play method can be invoked in Ruffle. I could not find a doc or example.
Here is my current code:
`var Rplayer;
window.RufflePlayer = window.RufflePlayer || {};
window.addEventListener("load", (event) => {
  const ruffle = window.RufflePlayer.newest();
  window.Rplayer = ruffle.createPlayer();
  const container = document.getElementById("peliCont");
  container.appendChild(Rplayer);
  Rplayer.load("gel_electrophoresis.swf").then(() => {
    console.info("Ruffle successfully loaded the file");
	}).catch((e) => {
    console.error("Ruffle failed to load the file: ${e}");
  });
});`

**Comment by SoftCreatR** [comment, marked as answer]
It works, I use it by myself to allow flash videos to be paused/resumed.

Simply call `window.Rplayer.pause()` or `window.Rplayer.play()` to make it work. However, it's currently hard to tell, if an instance is playing or paused. For this, I'm toggling a simple CSS class to determine it by myself. It works in most cases. The only exception I'm aware of is on page load, but that's fine, I guess. You may support my idea: https://github.com/ruffle-rs/ruffle/issues/3324

You can also check out our implementation here: https://www.softcreatr.com/z0r/

**Comment by angel-herraez** [comment]
Thank you. Unfortunately, it does not solve my problem with the particular swf file that I need to convert. It plays nicely with Ruffle and I can have pause / resume working as you say.
However, this animation is built in steps, after each one it stops and you need to click in a control inside the animation.  When it was Flash Player, the play() method worked the same as the click, so I could emulate the progress to next step using Javascript from a button in my webpage. However, in Ruffle this is not the case; the play() function does not make it progress to the next stage.
I guess this is not a problem with Ruffle, but with my swf file.
Anyway, thank you for explaining these options.
